### ISSUE TITLE
Setup without Mi Home (i.e. network setup) - possible?

### ISSUE THREAD

**Issue by Mazo** [samtap/fang-hacks#275]
Is it actually possible to set up a Xiaofang without having to resort to doing the initial setup with the Mi Home application? Would certainly be nice if you could just edit a few configuration files to set up networking and then not have to even look at the Mi Home app.

Does the Mi Home app actually do any configuration on the camera beyond initially setting networking settings?

**Comment by ril3y** [comment]
Yes,  I created a script that can configure the network without the app.
Msg me directly if you want to test it out.  I have not released it yet.

Riley

**Comment by Mazo** [comment]
@ril3y Would love to test it, but didn't github remove private messaging years ago?

**Comment by ril3y** [comment]
https://stackoverflow.com/a/44229207

Ha that worked on me but not on your account.

Riley

**Comment by bomellberg** [comment]
I second the wish to try this. Just got 4 cams home that needs configuring. I would love to not have to use the app at all.

**Comment by Mazo** [comment]
@ril3y Have sent an email to what I presume is your email address from the api.

**Comment by Mazo** [comment]
Looks like someone has already worked this out actually, pretty simple following the instructions in #132

@ril3y the QR code setup would still be a really helpful thing for a lot of users though!

**Comment by ril3y** [comment]
I just put it up here so it just "works" now.  It will get you online but your time will be wrong.  I didn't really feel like writing up all of the code to set that all right :)
Check it out and give it a go.  I made a video to show people the "steps".
https://codepen.io/ril3y/full/gXyzmO/

**Comment by bomellberg** [comment]
I can confirm that the QR generator is working. I just setup a camera in under two minutes. Start cam, press button, scan code, wait 20 sec, insert card, apply hacks, set timezone. Done!

I use motion + motioneye for supervision. Works great!

/Bo

**Comment by Mazo** [comment]
This issue can now be closed. I've added a wiki entry with instructions on both methods for setting WiFi details without Mi Home.

https://github.com/samtap/fang-hacks/wiki/Connecting-to-WiFi-without-using-Mi-Home
